### PR TITLE
Add § and ± as possible accelerators

### LINKED ISSUE #8208: Add § and ‡ keycodes to Electron Accelerator
Hi,

Just looking at the available keycodes for [Electron Accelerator](https://github.com/electron/electron/blob/master/docs/api/accelerator.md) and I could really use `§` (grave accent - 192) key support for something personal.

Would this be possible?

### LINKED ISSUE #8263: Accept § as a valid accelerator
The UK keyboard has this `§` key right under the `esc` key. Would be great to be able to use it for shortcut commands.

Does it make sense?

BTW: It's shifted value is `±`.

### PR DESCRIPTION
**Pull request by gfpacheco**
Fixes #8208

### CODE REVIEW

**Comment by kevinsawicki** [comment]
Just curious, what keyboard layout are you using where there are physical keys?

File: atom/common/keyboard_util.cc [hunk 1]
```diff
@@ -76,6 +76,7 @@ ui::KeyboardCode KeyboardCodeFromCharCode(base::char16 c, bool* shifted) {
     case '|': *shifted = true; case '\\': return ui::VKEY_OEM_5;
     case '}': *shifted = true; case ']': return ui::VKEY_OEM_6;
     case '"': *shifted = true; case '\'': return ui::VKEY_OEM_7;
+    case '±': *shifted = true; case '§': return ui::VKEY_OEM_8;
```
Comment (kevinsawicki) [inline atom/common/keyboard_util.cc]: This is failing to build with a compiler error:

```
../../atom/common/keyboard_util.cc:79:10: error: character too large for enclosing character literal type
    case '±': *shifted = true; case '§': return ui::VKEY_OEM_8;
         ^
../../atom/common/keyboard_util.cc:79:38: error: character too large for enclosing character literal type
    case '±': *shifted = true; case '§': return ui::VKEY_OEM_8;
```

See https://travis-ci.org/electron/electron/jobs/185833672 for more details.

**Comment by gfpacheco** [comment]
UK Apple keyboards have these

**Comment by zeke** [comment]
If this change is going to land, it should be documented in https://github.com/electron/electron/blob/master/docs/api/accelerator.md

**Comment by gfpacheco** [comment]
I thought it would be implicit in the "Punctuations like `~`, `!`, `@`, `#`, `$`, etc.".

How do you suggest me to document it?

**Comment by zeke** [comment]
I suppose you could just add it to that comma-separated list.

File: atom/common/keyboard_util.cc [hunk 2]
```diff
@@ -76,6 +76,9 @@ ui::KeyboardCode KeyboardCodeFromCharCode(base::char16 c, bool* shifted) {
     case '|': *shifted = true; case '\\': return ui::VKEY_OEM_5;
     case '}': *shifted = true; case ']': return ui::VKEY_OEM_6;
     case '"': *shifted = true; case '\'': return ui::VKEY_OEM_7;
+    
+    // ± and § respectively (from Apple UK keyboard)
+    case '\xB1': *shifted = true; case '\xA7': return ui::VKEY_OEM_8;
```
Comment (kevinsawicki) [inline atom/common/keyboard_util.cc]: Now this is generating the following warning:

```
ninja: Entering directory `out/D'
[3/6] CXX obj/atom/common/electron_lib.keyboard_util.o
../../atom/common/keyboard_util.cc:81:40: warning: overflow converting case value to switch condition type (-89 to 65447) [-Wswitch]
    case '\xB1': *shifted = true; case '\xA7': return ui::VKEY_OEM_8;
                                       ^
../../atom/common/keyboard_util.cc:81:10: warning: overflow converting case value to switch condition type (-79 to 65457) [-Wswitch]
    case '\xB1': *shifted = true; case '\xA7': return ui::VKEY_OEM_8;
```

**Comment by kevinsawicki** [comment]
@gfpacheco were you able to register a global shortcut for this?

I tried the following and it fails to register:

```js
const {app, globalShortcut} = require('electron')

app.once('ready', () => {
   const ret = globalShortcut.register('Control+§', () => {
    console.log('Control+§ is pressed')
  })

  if (!ret) {
    console.log('registration failed')
  }
})
```

**Comment by zcbenz** [comment]
I'm closing this since it has not activity for a long time, I'll reopen if you still want to work on this.

**Pull request closed without merging**